### DISCUSSION TITLE
Enable Auto Completions

### DISCUSSION THREAD

**Discussion by abdullahazmy** [community/community#131400] [Copilot Conversations] [answered]
### Enable Auto Completions

How can I disable the "Enable Auto Completions" feature for a specific language in VSCode?

I've tried editing settings.json and adjusting the settings provided by the extension, specifying the item and its value, but auto-completion is still active for C++. Any assistance would be appreciated.

![image](https://github.com/community/community/assets/125517573/e67d91d4-22bf-44d9-a8ef-f3a766ecfd25)

**Comment by Akash1134** [comment]
Hi @abdullahazmy ,

Are you asking this question in reference to GitHub Copilot or the auto-completion feature in VSCode? Copilot itself doesn't provide language-specific settings for auto-completions directly within GitHub.

Thanks!

**Reply by abdullahazmy** [reply]
Yes i meant GitHub Copilot.

**Reply by Akash1134** [reply, marked as answer]
As I mentioned, Copilot itself doesn't provide language-specific settings (like for C++ in your case) for disabling auto-completions directly - you could disable/enable auto-completion all together. You must change the settings.json in VSCode for a specific language. Reiterating the steps that needs to be followed:

1. **Open Settings**: Press `Ctrl + ,` to open the settings.
2. **Search for C++ settings**: In the search bar, type `C++`.
3. **Edit `settings.json`**: Click on the `Edit in settings.json` link.
4. **Add the following configuration**:

```json
"[cpp]": {
    "editor.quickSuggestions": {
        "other": false,
        "comments": false,
        "strings": false
    },
    "editor.suggestOnTriggerCharacters": false,
    "editor.acceptSuggestionOnEnter": "off",
    "editor.wordBasedSuggestions": false
}
```

This will disable auto-completion for C++ files in VS Code. If you continue to experience issues, consider checking if there are any conflicting extensions or settings.

**Reply by abdullahazmy** [reply]
I got it, thanks for your help.
Hope if Copilot allows this option soon, it will be too helpful.
